### PR TITLE
Clarify and expand API response filtering doc

### LINKED ISSUE #1424: Improve API response filtering docs
Following an excellent first version (https://docs.sensu.io/sensu-go/5.6/api/overview/#filtering), review and clarify API filtering docs.

To do:

- [x] Collect feedback
- [x] Review https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/
- [ ] Add a note in individual endpoint reference tables where filtering is available and link to API overview
- [x] Clarify use of different array syntaxes: `linux in check.subscriptions` vs. `check.namespace in [dev,production]`

User story reference:

~1. As a User, I want to filter Events by one or more Object labels to construct an Aggregate for evaluation (e.g. are >= 70% web apps up?)~ (handled in Aggregates guide/asset https://github.com/sensu/sensu-docs/pull/2162)
2. As a User, I want to filter Events by Entity
3. As a User, I want to filter Events by Check
4. As a User, I want to filter Events by Silenced
5. As a User, I want to filter Events by Severity (ok, warning, critical, unknown
6. As a User, I want a list of Events that identify an incident (check state failing)
8. As a User, I want to filter Events, Entities, and Checks by one or more subscriptions

### PR DESCRIPTION
**Pull request by hillaryfraley**
## Description
- Reorganize and revise Response filtering section content in API overview

## Motivation and Context
Closes https://github.com/sensu/sensu-docs/issues/1424

## Need to add:
- [x] Note in individual endpoint reference tables where filtering is available and link to API overview
- [x] As a User, I want to filter Events by Entity
- [x] As a User, I want to filter Events by Check
- [x] As a User, I want to filter Events by Severity (ok, warning, critical, unknown
- [x] As a User, I want a list of Events that identify an incident (check state failing -- anything that is not `0`)
- [x] As a User, I want to filter Events, Entities, and Checks by one or more subscriptions

## Review notes
There may be a bug in the silenced field selectors, so I left an example for silenced/unsilenced checks out for now. I filed a sensu-go issue: https://github.com/sensu/sensu-go/issues/3553

An associated concern is that we think operators will want to filter for silenced/unsilenced events, and this selector is not currently available. Cameron filed a sensu-go issue (https://github.com/sensu/sensu-go/issues/3552). When that selector is available and the question of a silenced bug is resolved, I can address docs issue https://github.com/sensu/sensu-docs/issues/2170 to update this page to include silenced/unsilenced examples.

### CODE REVIEW

File: content/sensu-go/5.17/api/assets.md [hunk 1]
```diff
@@ -77,7 +77,8 @@ HTTP/1.1 200 OK
 ---------------|------
 description    | Returns the list of assets.
 example url    | http://hostname:8080/api/core/v2/namespaces/default/assets
-pagination     | This endpoint supports pagination using the `limit` and `continue` query parameters. See the [API overview][2] for details.
+ppagination     | This endpoint supports [pagination][2] using the `limit` and `continue` query parameters.
```
Comment (rgeniesse) [inline content/sensu-go/5.17/api/assets.md]:
```suggestion
pagination     | This endpoint supports [pagination][2] using the `limit` and `continue` query parameters.
```

File: content/sensu-go/5.17/api/overview.md [hunk 2]
```diff
@@ -294,50 +292,167 @@ Here's the list of available fields:
...
+For example, to retrieve checks with a `linux` subscription:
 
 {{< highlight shell >}}
 curl -H "Authorization: Bearer $SENSU_ACCESS_TOKEN" http://127.0.0.1:8080/api/core/v2/checks -G \
---data-urlencode 'fieldSelector=check.publish == true && slack in check.handlers'
+--data-urlencode 'linux in check.subscriptions'
 {{< /highlight >}}
 
-In addition to selectors with multiple statements, you can use field and label selectors at the same time:
+_**NOTE**: The `in` and `notin` operators only work when the underlying value is a string. You cannot use `in` or `notin` with integer, float, array, or Boolean values._
```
Comment (rgeniesse) [inline content/sensu-go/5.17/api/overview.md]: May want to be more specific here on which value has to be a string, to the left of `in|notin`. As well as the item on the right needing to be an array. So `string [in|notin] array`.

**Comment by hillaryfraley** [comment]
@rgeniesse when this one is 👍 I will propagate the changes back to 5.14.

**Review by rgeniesse — CHANGES_REQUESTED** [review]
Just have the one comment on a note to maybe add more details.

Other than the, the examples will be huge for users to start creating their own filters expressions. And we can then use those examples to expand upon the dashboard and sensuctl references around this topic. 💯

File: content/sensu-go/5.17/api/overview.md (hunk 2 above)
Comment (hillaryfraley) [reply content/sensu-go/5.17/api/overview.md]: Yes, good call. I completely forgot to work that in. Thank you!

File: content/sensu-go/5.17/api/overview.md [hunk 3]
```diff
@@ -294,50 +292,167 @@ Here's the list of available fields:
...
+
+To retrieve checks that do not use the `slack` handler:
 
 {{< highlight shell >}}
 curl -H "Authorization: Bearer $SENSU_ACCESS_TOKEN" http://127.0.0.1:8080/api/core/v2/checks -G \
---data-urlencode 'fieldSelector=slack in check.handlers' \
---data-urlencode 'labelSelector=region != "us-west-1"'
+--data-urlencode 'slack notin check.handlers'
 {{< /highlight >}}
```
Comment (hillaryfraley) [inline content/sensu-go/5.17/api/overview.md:352]:
```suggestion
The `in` and `notin` operators have two important conditions.
First, `in` and `notin` only work when the underlying value you're filtering for is a string.
You can filter for strings and arrays of strings with `in` and `notin`, but you cannot use the `in` and `notin` operators to filter for integer, float, array, or Boolean values.

Second, if you're filtering for a string, the string must be to the **left** of the operator: `string [in|notin] selector`.
If you're filtering for an array of strings, the array must be to the **right** of the operator: `selector [in|notin] [string1,string2]`.
```

File: content/sensu-go/5.17/api/overview.md (hunk 2 above)
Comment (hillaryfraley) [inline content/sensu-go/5.17/api/overview.md]:
```suggestion
```

Comment (hillaryfraley) [reply content/sensu-go/5.17/api/overview.md]: I decided to move this explanation down to follow the `notin` example.

File: content/sensu-go/5.17/api/overview.md [hunk 4]
```diff
@@ -294,50 +292,166 @@ Here's the list of available fields:
...
+For example, to retrieve checks with a `linux` subscription:
 
 {{< highlight shell >}}
 curl -H "Authorization: Bearer $SENSU_ACCESS_TOKEN" http://127.0.0.1:8080/api/core/v2/checks -G \
---data-urlencode 'fieldSelector=check.publish == true && slack in check.handlers'
+--data-urlencode 'linux in check.subscriptions'
 {{< /highlight >}}
 
-In addition to selectors with multiple statements, you can use field and label selectors at the same time:
+
```
Comment (hillaryfraley) [inline content/sensu-go/5.17/api/overview.md]:
```suggestion
```

**Pull request merged**